nbs/04_data.external.ipynb

```diff
@@ -258,7 +258,7 @@
    "source": [
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #97.            local_path = URLs.LOCAL_PATH/('models' if c_key=='models' else 'data')/fname
</pre><p> </p><p>As I see this code only downloads or <code>models</code>  to <code>URLs.LOCAL_PATH/models</code>  if key is models then else downloads.</p><p> </p><p>Maybe we should double check it? just to see what happens when you call it with those params.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/muellerzr/fastai-docment-sprint/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Reply:
<p>Performed the following:</p><p> </p><ol><li>Current working director is empty (path pointed by <code>LOCAL_PATH = Path.cwd()</code>  )  </li><li>Run <code>URLs.path()</code>  with different values for <code>c_key</code> ['data', 'archive', 'storage', 'model']</li></ol><pre class="ql-syntax" spellcheck="false">for k in ['data', 'archive', 'storage', 'model']:
    print(URLs.path(URLs.MNIST_TINY,c_key=k))

"""
Output:

/root/.fastai/data/mnist_tiny.tgz
/root/.fastai/archive/mnist_tiny.tgz
/root/.fastai/tmp/mnist_tiny.tgz
/root/.fastai/models/mnist_tiny.tgz
"""
</pre><p> </p><p>Then,</p><ol><li> Copied the <code>mnist_tiny.tgz</code>  to my current director under folders <code>data</code>  and <code>models</code>  </li><li>Ran the same code</li></ol><pre class="ql-syntax" spellcheck="false">for k in ['data', 'archive', 'storage', 'model']:
    print(URLs.path(URLs.MNIST_TINY,c_key=k))

"""
Output:

/content/data/mnist_tiny.tgz
/content/data/mnist_tiny.tgz
/content/data/mnist_tiny.tgz
/content/data/mnist_tiny.tgz
"""
</pre><p> </p><p>So, this function can be called with one of ['data', 'archive', 'storage', 'model'], I think.</p><p> </p><p> </p><p> </p><p>PS: On a different note, there seems to be a bug in code in line 97:</p><p>local_path = URLs.LOCAL_PATH/('models' if c_key==<strong><em>'model<s>s</s></em></strong>' else 'data')/fname</p><p> </p><p>I think it should be <code>c_key == 'model'</code>  and NOT <code>c_key == 'models'</code>  (no <code>s</code>  at the end). I can create a separate PR for that</p>

Reply:
I think it's set to 'models' because that's the default directory for loading and saving in `Learner` (`model_dir='models'`):

```python
class Learner(GetAttr):
    _default='model'
    def __init__(self, dls, model, loss_func=None, opt_func=Adam, lr=defaults.lr, splitter=trainable_params, cbs=None,
                 metrics=None, path=None, model_dir='models', wd=None, wd_bn_bias=False, train_bn=True,
                 moms=(0.95,0.85,0.95)):
```
https://github.com/fastai/fastai/blob/ac905f74297296fd0d3a15282104bd33e978b4fc/fastai/learner.py#L85

Reply:
On second thought, I think you are correct and it should be `'models' if c_key=='model' else 'data'`, as `c_key` is for the `fastai_cfg` dictionary, not the folder.

Reply:
Okay, I can create a PR in the `fastai` repo and see how it goes